### DISCUSSION TITLE
Restricting user input until audio stimulus has played

### DISCUSSION THREAD

**Discussion by amunn** [jspsych/jsPsych#924] [Q&A] [answered]
I'm using the audio slider response plugin but I'd like to force participants to listen to the whole audio clip before allowing them to make judgement. Is there a way to disallow input until the audio has fully played?

**Comment by EceKayaMPI** [comment]
I modified my audio-keyboard-response plugin to restrict terminating the trial after the keypress (it waits for the stimuli to end). 
Would that help or do you also want it to ignore all inputs before the end of stimuli and wait for new input to end trial?

**Reply by amunn** [reply]
I don't think that would help. I would like to restrict input entirely until the audio has played.  This is mainly to prevent people from just clicking through the task without listening. To some extent this can be dealt post-hoc by using the rt per trial but it would be nice to control it somewhat during the experiment.

**Reply by EceKayaMPI** [reply]
If your stimuli has a fixed duration, you can add these lines and edit the plugin accordingly:
If you need the RT value just after the audio ends, you can add consecutive trials for each task. 

`
// wait for response AND trial duration to end trial 
      if (trial.response_ends_trial) {
        if(response.rt > trial.trial_duration){
        end_trial();
        } else {
          jsPsych.pluginAPI.setTimeout(function(){
            end_trial();
          }, Math.round(trial.trial_duration - response.rt) );
        }
      }
    };
`

Cheers,
Ece

**Reply by amunn** [reply]
Thanks @EceKayaMPI. I'll play around with that. The stimuli aren't of fixed duration, but it wouldn't be hard to add a duration value for each item to the stimulus file and use that. Can you expand a bit on your second comment about consecutive trials? I'm not not sure I understand what you mean there.

**Comment by jodeleeuw** [comment, marked as answer]
It's not currently a parameter in the plugin. That's something we can potentially add in the future. You could add it yourself by modifying the plugin file to either disable or hide the submit button until the audio finishes playing. There's already an event handler set up in the plugin for the `trial_ends_after_audio` parameter, so it wouldn't be too difficult to hook into that.

A second option, and what I would do, is to use one trial to play the audio and another to get the response. You could use `audio-keyboard-response` to play the file with `trial_ends_after_audio: true` and `choices: jsPsych.NO_KEYS` to end the trial immediately after the audio plays and not allow user input. Then you could use `html-slider-response` to prompt the user to respond. From the user's perspective, the audio will play and immediately after it finishes the response options will appear. It'll feel like a single trial to them.

**Reply by amunn** [reply]
@jodeleeuw Thanks, yes, your second method looks like it should do what I want. Now I think I understand @EceKayaMPI 's comment about consecutive trials better too.

**Reply by amunn** [reply]
Thanks @jodeleeuw  This second  method works perfectly for me.
